### DISCUSSION TITLE
How to concatenate files using Jaffree ( "-f concat -i <file list file> -c copy")

### DISCUSSION THREAD

**Discussion by daltontf** [kokorin/Jaffree#252] [Q&A] [answered]
Great library, I'm beginning to use on personal project where I extract cropped snippets from various files and then concatenate them in a final video. I can get the snippets generated and add them to a text input file, but having problems getting a concatenation done via Jaffree. The command line looks like:

ffmpeg -f concat -i concat_files.txt -c copy concatenated.mp4

Thanks,

**Comment by kokorin** [comment]
Could you clarify what problems you have with Jaffree? 

Does the command which you provided work from command line?

**Reply by daltontf** [reply]
I am working on a tool that crops and zooms clips from multiple videos and then concatenates the generated clip files together. Jaffree works great to extract the clips, but have trouble getting the concatenation to work. 

Currently using a command line: "ffmpeg -f concat -i concat_files.txt -c copy concatenated.mp4" with the "concat_files.txt" being a list of clip files like below:
file clip1.mp4
file clip2.mp4

Trying to figure out a way to make Jaffree do this. Thanks.

**Reply by kokorin** [reply]
There are several examples how to use Jaffree in README, also there are links to whole examples. So I would you suggest to start with that.
Also I'm not 100% sure, but AFAIK it's not possible to concatenate MP4 with `-c copy` option.

**Reply by daltontf** [reply]
The command line works for me, but it would be nice if I could do everything using Jaffree. Thanks

**Reply by kokorin** [reply]
Sorry, but I don't understand what the problem is. It is possible with Jaffree. Please check the examples. If something doesn't work, please provide code snippet that doesn't work.

**Comment by daltontf** [comment, marked as answer]
This works:

FFmpeg.atPath()
            .addInput(
                UrlInput.fromUrl(concatFile.getAbsolutePath())
                    .addArguments("-f", "concat")
                    .addArguments("-safe", "0")
            )
            .addArguments("-c", "copy")
            .setOverwriteOutput(true)
            .addOutput(UrlOutput.toUrl(new File(dir, "concatenated.mp4").getAbsolutePath()))
            .execute();
